### PR TITLE
Use constants for Frontend JS

### LINKED ISSUE #51: Use constants for configuration in JSX
We are currently exposing those as a configuration object, which – also never happening – could technically be altered during runtime. This limits the optimisations compilers can do in removing code. If those were constants, however, the compiler can safely assume they'd never changed and remove code that is behind a if (config.option){ ... } from execution time, similar as we already do it for the __DEBUG__-flag.

### PR DESCRIPTION
**Pull request by ligthyear**
Switching the config in the frontend code from a mutable object into exposing
all keys as `const` (constants), thus allowing the JS optimizers to remove code
that is never used/reached because of the app configuration.

Fixes #51

### CODE REVIEW

File: .infrastructure/webpack/make-webpack-config.js [hunk 1]
```diff
@@ -1,15 +1,40 @@
...
+var appConfig = merge(
+	{},
+	yaml.safeLoad(fs.readFileSync("beavy/config.yml"))[BEAVY_ENV],
+	yaml.safeLoad(fs.readFileSync('config.yml'))
+);
+
+function deepTransform(prefix, result, value, key) {
+	if (isObject(value))
+		transform(value,
+		  				partial(deepTransform, prefix + "_" + key),
```
Comment (ezmiller) [inline .infrastructure/webpack/make-webpack-config.js]: @ligthyear is the indentation off here?

File: .infrastructure/webpack/make-webpack-config.js [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
+			  			result);
+	if (!isBoolean(value))
+		value = JSON.stringify(value);
```
Comment (ezmiller) [inline .infrastructure/webpack/make-webpack-config.js]: And here too an indentation problem?

Comment (ligthyear) [inline .infrastructure/webpack/make-webpack-config.js]: Going for: #58 should do that. I will rebase this PR after merging 58 into master.

Comment (ligthyear) [inline .infrastructure/webpack/make-webpack-config.js]: actually #58 missed those files. Please review #59, which does!

**Comment by ligthyear** [comment]
@ezmiller giving this back to you for review.

**Comment by ezmiller** [comment]
@homu r+

**Pull request merged**